Feature/add export

JaccoGoris opened the pull request:
Added the ability to extract a 3D Collada file from the scene.
Reworked the way the walls are built because the way it was done was causing a lot of shading issues in other 3D software

Fix for #167

Commit:
aded collada exporter

Commit:
added export button using collada files and optimised the walls for export

Commit:
made build 2.1.0

JaccoGoris commented:
We moved our fork to the organisation, that's why this one is coming from "around media"

Commit:
swapped for newest threejs build with jsm exporter

JaccoGoris commented:
Updated my PR since the newest version of three allows the use of imports for the examples (finally) this is necessary because else the planner could not be used as a lib anymore.

danilosalvati commented on src/components/toolbar/toolbar-export-button.jsx:
```diff
@@ -0,0 +1,66 @@
+    // LOAD DATA
+    let planData = parseData(scene, actions, catalog);
+
+    setTimeout(() => {
```
This is not safe, it leads to bugs when the planData is not fully created (during my tests I was able to export an empty scene)

danilosalvati commented on src/catalog/factories/wall-factory-3d.js:
```diff
@@ -5,11 +5,12 @@ import {
+import { verticesDistance } from '../../utils/geometry';
 import * as SharedStyle from '../../shared-style';
 
 const halfPI = Math.PI / 2;
```
Not really necessary as you could define your own wall in the catalog

JaccoGoris replied:
this was just auto formatting I think, from saving the file again. I didn't change anything here.

danilosalvati commented on package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "react-planner",
-  "version": "2.0.5",
+  "version": "2.1.0",
```
I prefer to maintain the control over the versioning process (for example in this pull requests there are a few bugs so I would avoid publishing a release). So in this case it was preferable to let me merge the pull request and only after completing the process let me release the new version

JaccoGoris replied:
No problem at all!

danilosalvati commented:
Sorry for the late response but I was very busy in these days... I am not fully convinced by your pull requests, so I will try to accomplish same results with little modifications of your code base... In particular I was able to obtain an empty collada scene as the setTimeout was invoked before the scene parsing was complete

JaccoGoris commented:
@danilosalvati, No problem at all!
Since then I had found a couple of the same issues, and I've reworked it a bit already, there's a timeout now and I've changed to logic to work as an Action, rather than putting the functionality straight in the export button. it works much better now.

Though, for my tests when using it in a create react app: having the texture in the mtl file is problematic as it doesn't find it and the export fails. so that might have to change in the future.

do you still want me to update my PR or do you want to try yourself?

danilosalvati commented:
I have an idea I would like to test... so I think I will try myself for now

JaccoGoris commented:
Okay! very well :)

ericvanular commented:
@danilosalvati thanks for your work on this project. Can you please merge some of the awesome features @JaccoGoris has built here? Alternatively, the project will likely continue on a new fork
